compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/window/Dialog.skiko.kt

```diff
@@ -102,24 +120,71 @@ actual fun Dialog(
     } else {
         null
     }
-    PopupLayout(
-        popupPositionProvider = WindowCenterPositionProvider,
-        focusable = true,
+    DialogLayout(
         modifier = modifier,
         onOutsidePointerEvent = onOutsidePointerEvent,
+        properties = properties,
         content = content
     )
 }
 
-private object WindowCenterPositionProvider : PopupPositionProvider {
-    override fun calculatePosition(
-        anchorBounds: IntRect,
-        windowSize: IntSize,
-        layoutDirection: LayoutDirection,
-        popupContentSize: IntSize
-    ): IntOffset = windowSize.center - popupContentSize.center
+@Composable
+private fun DialogLayout(
+    properties: DialogProperties,
+    modifier: Modifier = Modifier,
+    onOutsidePointerEvent: ((PointerInputEvent) -> Unit)? = null,
+    content: @Composable () -> Unit
+) {
+    /*
+     * Keep empty layout as workaround to trigger layout after remove dialog.
+     * Required to properly update mouse hover state.
+     */
+    EmptyLayout()
+    RootLayout(
+        modifier = modifier,
+        focusable = true,
+        onOutsidePointerEvent = onOutsidePointerEvent
+    ) { owner ->
+        val measurePolicy = rememberDialogMeasurePolicy(properties) {
+            owner.bounds = it
+        }
+        Layout(
+            content = content,
+            measurePolicy = measurePolicy
+        )
+    }
+}
+
+@Composable
+private fun rememberDialogMeasurePolicy(
+    properties: DialogProperties,
+    onBoundsChanged: (IntRect) -> Unit
+) = remember(properties) {
+    MeasurePolicy { measurables, constraints ->
```

Comment:
Maybe reuse this layout between dialog and popup (and use "positioner" again)?

Reply:
It's already done in separate branch/next PR